web/src/main/java/co/mobileaction/example/web/controller/PostController.java

```diff
@@ -41,4 +42,26 @@ public ResponseEntity<Boolean> deletePost(@PathVariable Long postId)
 
         return ResponseEntity.ok(true);
     }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    public void createPost(@RequestBody Post post)
+    {
+        postService.savePost(post);
+    }
+
+    @DeleteMapping("/user/{userId}")
+    public ResponseEntity<Boolean> deletePostsOfUser(@PathVariable Long userId)
+    {
+        List<Post> userPosts = postService.findAllPostsOfUser(userId);
+        postService.deleteUserPosts(userPosts);

```

Comment:
this is ok, but for the performance concern, we don't need to fetch all the posts of users to the memory. Let's say there are 1M posts of a single user, when you fetch all the posts `postService.findAllPostsOfUser(userId)`, it will exceed the available memory. So, we can just use the delete query with userId parameter. Also, you need to think about using native query or not.